src/pyhf/pdf.py

```diff
@@ -464,6 +464,11 @@ def set_poi(self, name):
             raise exceptions.InvalidModel(
                 f"The parameter of interest '{name:s}' cannot be fit as it is not declared in the model specification."
             )
+        if self.param_set(name).n_parameters > 1:
```

Comment:
This now handles both `shapefactor` and `shapesys` if they act on channels with multiple bins. Both of these cases would fail the `assert` below. Setups like a `histosys` as POI seem to technically work, so even if they might not be useful in practice I guess there is no need to catch those here.